internal/app/singularity/cache_list_linux.go

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package singularity
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sylabs/singularity/internal/pkg/client/cache"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+func findSize(size int64) (string, error) {
+	var sizeF float64
+	if size <= 1000000 {
+		sizeF = float64(size) / 1000
+		return strings.Join([]string{fmt.Sprintf("%.2f", sizeF), " Kb"}, ""), nil
+	} else if size <= 1000000000 {
+		sizeF = float64(size) / 1000000
+		return strings.Join([]string{fmt.Sprintf("%.2f", sizeF), " Mb"}, ""), nil
+	} else if size >= 1000000000 {
+		sizeF = float64(size) / 1000000000
+		return strings.Join([]string{fmt.Sprintf("%.2f", sizeF), " Gb"}, ""), nil
+	}
+	return "", fmt.Errorf("failed to detect file size")
+}
+
+func listLibraryCache() error {
+	// loop thrught library cache
+	libraryCacheFiles, err := ioutil.ReadDir(cache.Library())
+	if err != nil {
+		sylog.Warningf("Unable to opening cache folder: %v", err)
+		return err
+	}
+	for _, f := range libraryCacheFiles {
+		cont, err := ioutil.ReadDir(filepath.Join(cache.Library(), f.Name()))
+		if err != nil {
+			sylog.Warningf("Unable to looking in cache: %v", err)
+			return err
+		}
+		for _, c := range cont {
+			fileInfo, err := os.Stat(filepath.Join(cache.Library(), f.Name(), c.Name()))
+			if err != nil {
+				sylog.Warningf("Unable to get stat: %v", err)
+				return err
+			}
+			printFileSize, err := findSize(fileInfo.Size())
+			if err != nil {
+				// no need to describe the error, since it is already
+				sylog.Warningf("%v", err)
+			}
+			fmt.Printf("%-22s %-22s %-16s %s\n", c.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "library")
+			printFileSize = ""
+		}
+	}
+	return nil
+}
+
+func listOciCache() error {
+	// loop thrught oci-tmp cache
+	ociTmp, err := ioutil.ReadDir(cache.OciTemp())
+	if err != nil {
+		sylog.Warningf("Unable to opening oci-tmp folder: %v", err)
+		return err
```

Comment:
In general, we should follow the pattern of returning the information with the error. So instead of printing the Warning message here (imagine a scenario where the user is running --quiet), we should instead do something like:

`return fmt.Errorf("while opening oci-tmp folder: %v", err)`

I think there are a few other places in this PR where you should make that change!